browser/src/map/handler/Map.KeyboardShortcuts.ts

```diff
@@ -26,21 +26,32 @@ enum Mod {
     SHIFT   = 4,
 }
 
+enum ViewType {
+    Edit = 0,
+    ReadOnly = 1,
+}
+
+type shortcutCallback = () => void;
+
 class ShortcutDescriptor {
     docType: string; // if undefined then all apps match
     eventType: string;
     modifier: Mod;
     key: string;
     unoAction: string;
     dispatchAction: string;
+    viewType: ViewType;
+    callback: shortcutCallback;
 
-    constructor(docType: string, eventType: string, modifier: Mod, key: string, unoAction: string, dispatchAction: string) {
+    constructor(docType: string, eventType: string, modifier: Mod, key: string, unoAction: string, dispatchAction: string, viewType: ViewType = ViewType.Edit | ViewType.ReadOnly, callback: shortcutCallback = null) {
         this.docType = docType;
         this.eventType = eventType;
         this.modifier = modifier;
         this.key = key;
         this.unoAction = unoAction;
         this.dispatchAction = dispatchAction;
+        this.viewType = viewType;
+        this.callback = callback;
```

Comment:
I would prefer to use `unoAction` for core uno commands and `dispatchAction` for Online JS custom callbacks only. So this list will be free from `code`

Reply:
Thanks for the comments :)